nvtabular/loader/backend.py

```diff
@@ -349,72 +312,80 @@ def create_tensors(self, gdf, cat_names=None, cont_names=None, label_names=None)
         raise NotImplementedError()
 
 
-class TensorBatchDatasetItrFactory:
-    """
-        This is the Factory that creates the correct iterator,
-        given a target library.
-
-        Parameters:
-        dataset: NVTabular dataset
-        target: str, target library you want the iterator supported - torch
-        shuffle: bool, shuffle dataset
-        device: int, represents targeted GPU id
-        total_devs: [int], list represents all avialable GPU IDs
-    """
-
-    def create(self, dataset, target, shuffle=False, device=None, total_devs=None, **kwargs):
-        if target in "torch":
-            return TorchTensorBatchDatasetItr(
-                dataset, shuffle=shuffle, device=device, total_devs=total_devs, **kwargs
-            )
-        else:
-            raise ValueError(target)
-
+def _validate_workflows(workflows, cat_names, cont_names, label_names):
+    assert all([isinstance(w, BaseWorkflow) for w in workflows])
+    # TODO: commenting out until it's clearer what the
+    # columns in workflow.columns_cts["final"]["ctx"] mean
+    # for workflow in workflows:
```

Comment:
here what you should be getting is a list of operator names (IDs) that then need to be looked up within the columns context for the columns prescribed to that operator.

Reply:
Hmmm ok yeah I see the operator names, so how would I use those to map to expected output column names?

Reply:
I think you are looking for this https://github.com/NVIDIA/NVTabular/blob/main/nvtabular/workflow.py#L383 that will take the columns context object and pulls the actual correct columns.

Reply:
Ok I guess that makes sense, so we need to be calling `Workflow.finalize()` and `Workflow.create_final_cols()` once a workflow is done be added to?

This is probably a discussion for elsewhere, but I wonder if it might make sense to implement `columns_ctx` as a `@property` and just create this on the fly whenever users need it, making it easier to inspect at intermediate steps during creation.